vm/src/stdlib/time.rs

```diff
@@ -173,6 +197,169 @@ fn time_strptime(
     Ok(PyStructTime::new(vm, instant, -1))
 }
 
+fn get_thread_time(vm: &VirtualMachine) -> PyResult<Duration> {
+    cfg_if::cfg_if! {
+        if #[cfg(windows)] {
+            let mut creation_time = FILETIME::default();
+            let mut exit_time = FILETIME::default();
+            let mut kernel_time = FILETIME::default();
+            let mut user_time = FILETIME::default();
+
+            unsafe {
+                let thread = GetCurrentThread();
+                if GetThreadTimes(thread, &mut creation_time, &mut exit_time, &mut kernel_time, &mut user_time) == 0 {
+                    return Err(vm.new_os_error("Failed to get clock time".to_owned()));
+                }
+
+                let mut large: ULARGE_INTEGER = std::mem::zeroed();
+                large.u_mut().LowPart = kernel_time.dwLowDateTime;
+                large.u_mut().HighPart = kernel_time.dwHighDateTime;
+                let ktime = *large.QuadPart();
+
+                large.u_mut().LowPart = user_time.dwLowDateTime;
+                large.u_mut().HighPart = kernel_time.dwHighDateTime;
+                let utime = *large.QuadPart();
+
+                Ok(Duration::from_nanos((ktime + utime) * 100))
+            }
+        } else if #[cfg(any(
+            target_os = "android",
+            target_os = "dragonfly",
+            target_os = "freebsd",
+            target_os = "linux",
+            target_os = "openbsd",
+        ))] {
+            let mut time = libc::timespec {
+                tv_sec: 0,
+                tv_nsec: 0,
+            };
+            if unsafe { libc::clock_gettime(libc::CLOCK_THREAD_CPUTIME_ID, &mut time) } == -1 {
+                return Err(vm.new_os_error("Failed to get clock time".to_owned()));
+            }
+            Ok(Duration::new(time.tv_sec as u64, time.tv_nsec as u32))
+        } else if #[cfg(solaris)] {
+            Ok(Duration::from_nanos(unsafe { libc::gethrvtime() }))
+        } else {
+            Err(vm.new_not_implemented_error("thread time unsupported in this system".to_owned()))
+        }
+    }
+}
+
+fn time_thread_time(vm: &VirtualMachine) -> f64 {
+    match get_thread_time(vm) {
+        Ok(v) => v.as_secs_f64(),
+        Err(err) => panic!("Time error: {:?}", err),
+    }
+}
```

Comment:
By looking inside `get_thread_time` the `Err` case looks able to be raised practically.

```suggestion
fn time_thread_time(vm: &VirtualMachine) -> PyResult<f64> {
    Ok(get_thread_time(vm)?.as_secs_f64())
}
```
And also other functions

Reply:
That way i can avoid panic call. Thanks